### PR TITLE
[Merged by Bors] - feat(Data/Sign): add missing API lemmas

### PR DESCRIPTION
**Pull request by MichaelStollBayreuth**
This adds three `simp` and `norm_cast` lemmas for `SignType`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Sign.lean [hunk 1]
```diff
@@ -286,6 +286,21 @@ theorem range_eq {α} (f : SignType → α) : Set.range f = {f zero, f neg, f po
...
+
+@[simp, norm_cast] lemma SignType.cast_pow {α} [MonoidWithZero α] [HasDistribNeg α]
+    (a : SignType) (k : Nat) :
+    ((a ^ k : SignType) : α) = (a : α) ^ k := by
+  simp_rw [← SignType.castHom_apply, map_pow]
+
+@[simp, norm_cast] lemma SignType.cast_zpow {α} [GroupWithZero α] [HasDistribNeg α]
+    (a : SignType) (k : Int) :
+    ((a ^ k : SignType) : α) = (a : α) ^ k := by
+  simp_rw [← SignType.castHom_apply, map_zpow₀]
```
Comment (eric-wieser) [inline Mathlib/Data/Sign.lean:302]:
```suggestion
@[simp, norm_cast] lemma SignType.cast_mul {α} [MulZeroOneClass α] [HasDistribNeg α]
    (a b : SignType) :
    ↑(a * b) = (a : α) * b :=
  map_mul SignType.castHom _ _

@[simp, norm_cast] lemma SignType.cast_pow {α} [MonoidWithZero α] [HasDistribNeg α]
    (a : SignType) (k : Nat) :
    ↑(a ^ k) = (a : α) ^ k :=
  map_pow SignType.castHom _ _

@[simp, norm_cast] lemma SignType.cast_zpow {α} [GroupWithZero α] [HasDistribNeg α]
    (a : SignType) (k : Int) :
    ↑(a ^ k) = (a : α) ^ k :=
  map_zpow₀ SignType.castHom _ _
```

Comment (MichaelStollBayreuth) [reply Mathlib/Data/Sign.lean]: Done (and removed double `SignType`).

File: Mathlib/Data/Sign.lean [hunk 2]
```diff
@@ -286,6 +286,20 @@ theorem range_eq {α} (f : SignType → α) : Set.range f = {f zero, f neg, f po
...
+
+@[simp, norm_cast] lemma cast_pow {α} [MonoidWithZero α] [HasDistribNeg α]
+    (a : SignType) (k : Nat) :
+    ↑(a ^ k) = (a : α) ^ k :=
+  map_pow SignType.castHom _ _
+
+@[simp, norm_cast] lemma cast_zpow {α} [GroupWithZero α] [HasDistribNeg α]
+    (a : SignType) (k : Int) :
+    ↑(a ^ k) = (a : α) ^ k :=
+  map_zpow₀ SignType.castHom _ _
```
Comment (eric-wieser) [inline Mathlib/Data/Sign.lean:301]: Why not use `\N` and `\Z`?
```suggestion
@[simp, norm_cast] lemma coe_mul {α} [MulZeroOneClass α] [HasDistribNeg α] (a b : SignType) :
    ↑(a * b) = (a : α) * b :=
  map_mul SignType.castHom _ _
@[simp, norm_cast] lemma coe_pow {α} [MonoidWithZero α] [HasDistribNeg α] (a : SignType) (k : ℕ) :
    ↑(a ^ k) = (a : α) ^ k :=
  map_pow SignType.castHom _ _
@[simp, norm_cast] lemma coe_zpow {α} [GroupWithZero α] [HasDistribNeg α] (a : SignType) (k : ℤ) :
    ↑(a ^ k) = (a : α) ^ k :=
  map_zpow₀ SignType.castHom _ _
```

Comment (eric-wieser) [reply Mathlib/Data/Sign.lean:301]: (also rename `cast` to `coe` to match the lemmas above)

Comment (MichaelStollBayreuth) [reply Mathlib/Data/Sign.lean:301]: Indeed. This actually saves two lines :smile:

Comment (MichaelStollBayreuth) [reply Mathlib/Data/Sign.lean:301]: OK to the second point as well.

**Comment by eric-wieser** [comment]
bors d+

Thanks!

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**